src/ng/services/table.spec.js

```diff
@@ -0,0 +1,92 @@
+import Table from './table';
+import angular from 'angular'; //necessary to test CellDom model property
+
+const myWindow = window.open("", "MsgWindow", "width=100,height=50");
+function htmlToElement(html) {
+	const template = myWindow.document.createElement('template');
+	template.innerHTML = html;
+	return template.content.firstChild;
+}
+
+describe('Table', () => {
+	const table = '<table tabindex=\"0\"><thead><tr><th><label aria-label=\"Last Name\">' +
+		'Last Name</label></div></th><th><label aria-label=\"First Name\">First Name</label></div></th></tr></thead><tbody><tr>' +
+		'<td class=\"q-grid-rowIndicator q-grid-row-indicator\">Laserna</td><td class=\"q-grid-rowIndicator q-grid-row-indicator\">' +
+		'Lue</td></tr><tr><td class=\"q-grid-rowIndicator q-grid-row-indicator\">Grebel</td><td class=\"q-grid-rowIndicator q-grid-row-indicator\">' +
+		'Jasper</td></tr><tr><td class=\"q-grid-rowIndicator q-grid-row-indicator\">Beichner</td>' +
+		'<td class=\"q-grid-rowIndicator q-grid-row-indicator\">Kandra</td></tr><tr><td class=\"q-grid-rowIndicator q-grid-row-indicator\">' +
+		'Schmith</td><td class=\"q-grid-rowIndicator q-grid-row-indicator\">Jeff</td></tr></tbody></table>';
+	const body = '<tbody><tr><td class=\"q-grid-rowIndicator q-grid-row-indicator\">Laserna</td>' +
+		'<td class=\"q-grid-rowIndicator q-grid-row-indicator\">Lue</td></tr><tr><td class=\"q-grid-rowIndicator q-grid-row-indicator\">' +
+		'Grebel</td><td class=\"q-grid-rowIndicator q-grid-row-indicator\">Jasper</td></tr><tr>' +
+		'<td class=\"q-grid-rowIndicator q-grid-row-indicator\">Beichner</td><td class=\"q-grid-rowIndicator q-grid-row-indicator\">' +
+		'Kandra</td></tr><tr><td class=\"q-grid-rowIndicator q-grid-row-indicator\">Schmith</td>' +
+		'<td class=\"q-grid-rowIndicator q-grid-row-indicator\">Jeff</td></tr></tbody>';
+	const head = '<thead><tr><th><label aria-label=\"Last Name\">Last Name</label></div></th><th><label aria-label=\"First Name\">' +
+		'First Name</label></div></th></tr></thead>';
+
+	const markup = {
+		table: htmlToElement(table),
+		body: htmlToElement(body),
+		head: htmlToElement(head)
+	};
+
+	const markupEmpty = {};
+	const tableArr = [new Table(markup), new Table(markupEmpty)];
+	const tableProps = ['body', 'head', 'foot'];
+
+	describe('table should have properties and they should be objects', () => {
```

Comment:
Actually there is no much sence to check if poperty is exists, all this properties should be covered by unit tests